apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/DataActivity.java

```diff
@@ -233,6 +240,26 @@ public void onNothingSelected(AdapterView<?> parent) {
         });
     }
 
+    private void fillIntentWithDocDetails(Intent intent, DocumentWrapper document, String partition) {
+        intent.putExtra(DOCUMENT_PARTITION, partition);
+        intent.putExtra(DOCUMENT_ID, document.getId());
+        if (document.getError() != null) {
+            String message = document.getError().getMessage();
+            if (message.length() > MAX_CONTENT_LENGTH) {
+                message = message.substring(0, MAX_CONTENT_LENGTH) + "...";
+                intent.putExtra(DOCUMENT_ERROR, message);
+                intent.putExtra(DOCUMENT_ERROR_NULL_STATUS, true);
+            }
+        } else {
+            intent.putExtra(DOCUMENT_DATE, document.getLastUpdatedDate().toString());
```

Comment:
date and state might be set on error.
Use state either way, and for date, check for null date instead.